xl/xenops_server_xenlight.ml

```diff
@@ -2995,9 +3021,11 @@ let process_one_watch xc xs (path, token) =
 	if path = _introduceDomain || path = _releaseDomain
 	then look_for_different_domains xc xs
 	else match List.filter (fun x -> x <> "") (Xstringext.String.split '/' path) with
-		| "local" :: "domain" :: domid :: "backend" :: kind :: frontend :: devid :: _ ->
+		| "local" :: "domain" :: domid :: "backend" :: kind :: frontend :: devid :: key ->
 			debug "Watch on backend domid: %s kind: %s -> frontend domid: %s devid: %s" domid kind frontend devid;
-			fire_event_on_device frontend kind devid
+			fire_event_on_device frontend kind devid;
+			(* If this event was a state change then this might be the first time we see evidence of PV drivers *)
+			if key = ["state"] then maybe_update_pv_drivers_detected ~xc ~xs (int_of_string frontend) path
```

Comment:
I was referring to the case where there are no VIFs or VBDs.

On Wed, 06 Apr 2016 at 06:25, Jonathan Davies wrote:
> >  			debug "Watch on backend domid: %s kind: %s -> frontend domid: %s devid: %s" domid kind frontend devid;
> > -			fire_event_on_device frontend kind devid
> > +			fire_event_on_device frontend kind devid;
> > +			(* If this event was a state change then this might be the first time we see evidence of PV drivers *)
> > +			if key = ["state"] then maybe_update_pv_drivers_detected ~xc ~xs (int_of_string frontend) path
> 
> By callback IRQ do you mean event channel? I don't think that adds anything. Being in state 4 means that the frontend is connected to the backend which implies that there's an event-channel set up and a shared-memory ring mapped.
> 
> ---
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub:
> https://github.com/xapi-project/xenopsd/pull/261/files/b7b076a6697875356a96c624943e0eec9a6b1b32#r58703394
Si.